homeassistant/components/wallbox/select.py

```diff
@@ -0,0 +1,104 @@
+"""Home Assistant component for accessing the Wallbox Portal API. The switch component creates a switch entity."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import (
+    CHARGER_DATA_KEY,
+    CHARGER_FEATURES_KEY,
+    CHARGER_PLAN_KEY,
+    CHARGER_POWER_BOOST_KEY,
+    CHARGER_SERIAL_NUMBER_KEY,
+    CHARGER_SOLAR_CHARGING_MODE,
+    DOMAIN,
+    EcoSmartMode,
+)
+from .coordinator import WallboxCoordinator
+from .entity import WallboxEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class WallboxSelectEntityDescription(SelectEntityDescription):
+    """Describes Wallbox select entity."""
+
+    current_option_fn: Callable[[WallboxCoordinator], str | None]
+    select_option_fn: Callable[[WallboxCoordinator, str], Awaitable[None]]
+
+
+SELECT_TYPES: dict[str, WallboxSelectEntityDescription] = {
+    CHARGER_SOLAR_CHARGING_MODE: WallboxSelectEntityDescription(
+        key=CHARGER_SOLAR_CHARGING_MODE,
+        translation_key="eco_smart",
+        options=[
+            EcoSmartMode.OFF,
+            EcoSmartMode.ECO_MODE,
+            EcoSmartMode.FULL_SOLAR,
+        ],
+        select_option_fn=lambda coordinator, mode: coordinator.async_set_eco_smart(
+            mode
+        ),
+        current_option_fn=lambda coordinator: coordinator.data[
+            CHARGER_SOLAR_CHARGING_MODE
+        ],
+    )
```

Comment:
```suggestion
    CHARGER_ECO_SMART_KEY: WallboxSelectEntityDescription(
        key=CHARGER_ECO_SMART_KEY,
        translation_key=CHARGER_ECO_SMART_KEY,
        options=[
            EcoSmartMode.OFF,
            EcoSmartMode.ECO_MODE,
            EcoSmartMode.FULL_SOLAR,
        ],
        select_option_fn=lambda coordinator, mode: coordinator.async_set_eco_smart(
            mode
        ),
        current_option_fn=lambda coordinator: coordinator.data[
            CHARGER_ECO_SMART_KEY
        ],
    )
```

if you accept the suggestion of using `CHARGER_ECO_SMART_KEY` in the coordinator, we can also change this one to make it consistent with the `sensor` platform